megamek/src/megamek/client/ratgenerator/MissionRole.java

```diff
@@ -418,6 +419,132 @@ public static Double adjustAvailabilityByRole(double avRating, Collection<Missio
 						avRating += avAdj[2] * (mRec.getSpeed() - (7 - mRec.getWeightClass()));					
 					}
 					break;
+				//Large craft roles
+				case CIVILIAN:
+				    if (!mRec.getRoles().contains(CIVILIAN)) {
+				        return null;
+				    }
+				    break;
+				case CARGO:
+				    if (mRec.getRoles().contains(CARGO)) {
+				        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(CIVILIAN)) {
+                        avRating += avAdj[1];
+                    } else {
+                        avRating -= avAdj[2];
+                    }
+                    break;
+				case SUPPORT:
+                    if (mRec.getRoles().contains(SUPPORT)) {
+                        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(TUG)) {
+                        avRating += avAdj[1];
+                    } else if (mRec.getRoles().contains(CARGO)
+                                || mRec.getRoles().contains(CIVILIAN)) {
+                        avRating += avAdj[0];
+                    } else {
+                        avRating -= avAdj[2];
+                    }
+                    break;
+				case TUG:
+                    if (!mRec.getRoles().contains(TUG)) {
+                        return null;
+                    }
+                    break;
+				case POCKET_WARSHIP:
+				    if (mRec.getRoles().contains(POCKET_WARSHIP)) {
+                        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(ASSAULT)) {
+                        avRating += avAdj[0];
+                    } else {
+                        return null;
+                    }
+				    break;
+				case ASSAULT:
+                    if (mRec.getRoles().contains(ASSAULT)) {
+                        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(POCKET_WARSHIP)) {
+                        avRating += avAdj[0];
+                    } else {
+                        return null;
+                    }
+                    break;
+				case MECH_CARRIER:
+				    if (mRec.getRoles().contains(MECH_CARRIER)) {
+                        avRating += avAdj[2];
+                    } else if (mRec.getRoles().contains(TROOP_CARRIER)) {
```

Comment:
TROOP_CARRIER is a combined forces transport, but it may just be vees and infantry. MECH_CARRIER should just be those with mech bays.

Reply:
I had done this intentionally because a number of the ships listed as troop carriers include (substantial) mech carrier capabilities -  Colossus, Excalibur...
I recognize that every now and again it'll return a ship with no mech carrying capacity. 

For myself, I've updated the forcegen files to reflect each individual role as well where this is the case, but since my files are customized I haven't submitted those changes to the repo.

Reply:
Ok, I'm good with that.